speechbrain/processing/features.py

```diff
@@ -130,13 +132,15 @@ def __init__(
 
         self.window = window_fn(self.win_length)
 
-    def forward(self, x):
+    def forward(self, x, wav_len=None):
         """Returns the STFT generated from the input waveforms.
 
         Arguments
         ---------
         x : tensor
             A batch of audio signals to transform.
+        wav_len: tensor
+            Relative length of each sentence in the batch.
```

Comment:
`relative length` is not very precise.  Also, the documentation for `x: tensor` is not very specific in terms of layout, is it, (batch_size, num_samples)?  And any constraints on the dtype?

Reply:
... e.g. is the relative length tensor a tensor of shape (batch_size,) containing the duration of each utterance divided by the maximum duration of any of the utterances?

Reply:
The relative length tensor is the lengths of the waveforms divided by the max length in the batch. E.g., if the lengths of wavs in a batch is [1000, 5000], we convert it to [0.2, 1.0].
`x: tensor` is expected to be a float tensor variable with the shape (batch_size, num_samples). I will specify this part; also, suggest using type hints (PEP 484) in the future release.

Reply:
Right,  we can be more specific in the docstring here. 

We indeed use relative lengths just for convenience/simplicity. 
There are operations that change the time resolution of the input waveform (e.g., `STFT`, `conv1`, `conv2d`, `pooling`, etc.). Keeping track of the absolute length after each operation is a bit annoying. For instance, many people use `torch.sequential `to create pipelines of computations, which requires to output exactly one tensor (not tensor + length).  With the relative length approach, we only need to have this information after the batch generation. For instance, if we create a batch with a sentence of length 16000 samples and one of 32000, the relative lengths vector will be `[0.5, 1.0]`.  After feature computation (or any other operation that changes the input resolution) we might have a batch of size let's say  200 time steps. The valid number of time steps for the first sentence is round(200 * 0.5) = 100. Normally, the exact number of time steps is used for excluding the zero-padded time steps from the loss computation or from statistical pooling (e.g, see xvector or ECAPA models).

Said that, it is true that the information might not always be 100% correct (e.g. +1, -1 steps depending on how the first/last windows are used). If there is a strong preference from the users for the absolute length approach, we can change it in the next version of SpeechBrain. We might wanna change the current container called L`enghtCapableSequential `to manage pipelines where we return the new absolute length if an operation like conv, pooling, etc is called. The most elegant approach would be to inject the length as a kind of meta information in the tensor. Not sure if this is possible though.